packages/fuels-core/src/json_abi.rs

```diff
@@ -526,12 +532,35 @@ pub fn parse_param(param: &Property) -> Result<ParamType, Error> {
                 }
                 return parse_array_param(param);
             }
+            if param.type_field.starts_with('(') && param.type_field.ends_with(')') {
+                // Try to parse tuple (T, T, ..., T)
+                return parse_tuple_param(param);
+            }
             // Try to parse enum or struct
             parse_custom_type_param(param)
         }
     }
 }
 
+pub fn parse_tuple_param(param: &Property) -> Result<ParamType, Error> {
```

Comment:
This should be unit-tested

Reply:
https://github.com/FuelLabs/fuels-rs/pull/205#discussion_r844102421.